designer/server/src/routes/forms/create.js

```diff
@@ -198,7 +200,14 @@ export default [
       // Submit new form metadata
       try {
         if (!result.error) {
-          await forms.create(result.value)
+          // @ts-expect-error - user is not undefined
```

Comment:
Sorry @davidjamesstone, can we remove this?

We can't guarantee that our code will always populate `credentials.user`

Let's use a tighter check from https://github.com/DEFRA/forms-designer/pull/148/files#r1595588500 instead